### PR TITLE
[quantization] Use ConvPackedParams everywhere

### PR DESCRIPTION
**Pull request by jamesr66a**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30199 [quantization] Use ConvPackedParams everywhere**

This makes things simpler and is probably how we should've done it in the first place

Differential Revision: [D18628004](https://our.internmc.facebook.com/intern/diff/D18628004)

### CODE REVIEW

File: torch/nn/quantized/modules/conv.py [hunk 1]
```diff
@@ -317,27 +339,24 @@ def _get_name(self):
 
     def set_weight_bias(self, w, b):
         # type: (torch.Tensor, Optional[torch.Tensor]) -> None
-        self._packed_params = torch.ops.quantized.conv3d_prepack(
-            w, b, self.stride, self.padding, self.dilation, self.groups)
+        self._packed_params.set_weight_bias(w, b)
 
     def _weight_bias(self):
-        return torch.ops.quantized.conv3d_unpack(self._packed_params)
+        return self._packed_params._weight_bias()
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/conv.py]: this is conv2d_unpack....we'll need to have Conv3dPackedParams module for this..

Comment (jamesr66a) [reply torch/nn/quantized/modules/conv.py]: Are there any tests or checks for this? It all passes just fine in the test suite....

**Comment by jerryzh168** [comment]
btw LinearPackedParams and ConvPackedParams are temporary workaround in graphmode to support serialization of packed values. And long term support would be TorchBind. So what about eager mode? do we always need to do this workaround?

File: torch/nn/quantized/modules/conv.py (hunk 1 above)
Comment (jerryzh168) [reply torch/nn/quantized/modules/conv.py]: cc @dskhudia @BIT-silence do we have tests for conv3d_prepack and conv3d_unpack?

Comment (jamesr66a) [reply torch/nn/quantized/modules/conv.py]: @jerryzh168 figured out the missing test coverage, please see https://github.com/pytorch/pytorch/pull/30247

**Comment by jamesr66a** [comment]
@jerryzh168 this is the correct way to do it, I've already done the same refactoring for RNNs. We can upgrade to using torchbind in the future, but after this change that should just be a 1:1 replacement of the TorchScript *PackedParams modules with the torchbind class

File: torch/nn/quantized/modules/conv.py [hunk 2]
```diff
@@ -16,10 +16,103 @@
 from torch.nn.quantized.modules.utils import _quantize_weight
 from torch.nn.utils import fuse_conv_bn_weights
 
+
+class ConvPackedParams(torch.nn.Module):
+    def __init__(self, spatial_dim):
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/conv.py]: maybe make default value 2?

Comment (jamesr66a) [reply torch/nn/quantized/modules/conv.py]: I don't think a default value saves us much here, and it is bound to cause trouble when others are trying to use this class

Comment (jerryzh168) [reply torch/nn/quantized/modules/conv.py]: I see, makes sense

**Review by jerryzh168 — APPROVED** [review]
LGTM

File: torch/nn/quantized/modules/conv.py (hunk 1 above)
Comment (dskhudia) [reply torch/nn/quantized/modules/conv.py]: > cc @dskhudia @BIT-silence do we have tests for conv3d_prepack and conv3d_unpack?

Yes. There is.

**Comment by raghuramank100** [comment]
Can you also add versioning support like was done in https://github.com/pytorch/pytorch/pull/30481 as part of this PR?

**Review by raghuramank100 — DISMISSED** [review]
Please include support for older versions of the state dict as part of this PR so that torchvision support for quantized models is not broken (just like https://github.com/pytorch/pytorch/pull/30481)

**Pull request closed without merging**